subprojects/platform-native/src/main/java/org/gradle/nativeplatform/toolchain/internal/gcc/GccPlatformToolProvider.java

```diff
@@ -182,8 +198,17 @@ private String getPCHFileExtension() {
         return ".h.gch";
     }
 
+    private GccMetadata getGccMetadata(ToolType compilerType) {
+        GccCommandLineToolConfigurationInternal compiler = toolRegistry.getTool(compilerType);
+        CommandLineToolSearchResult searchResult = toolSearchPath.locate(compiler.getToolType(), compiler.getExecutable());
+        String language = LANGUAGE_FOR_COMPILER.get(compilerType);
+        List<String> languageArgs = language == null ? Collections.<String>emptyList() : ImmutableList.of("-x", language);
+        return metadataProvider.getCompilerMetaData(searchResult.getTool(), languageArgs);
+    }
+
     @Override
-    public List<File> getSystemIncludes() {
+    public List<File> getSystemIncludes(ToolType compilerType) {
+        GccMetadata gccMetadata = getGccMetadata(compilerType);
```

Comment:
I'd love to see a unit test around this that proves that when I see the ToolType for each of the LANGUAGE_FOR_COMPILER types, I see a call to get a compiler metadata from `metadataProvider` with the correct languageArgs.  Seems like you should be able to stub everything out and do this pretty easily.

Reply:
I added the test.